core/src/main/java/org/apache/iceberg/encryption/StandardEncryptionManager.java

```diff
@@ -67,7 +72,15 @@ public InputFile decrypt(EncryptedInputFile encrypted) {
   @Override
   public Iterable<InputFile> decrypt(Iterable<EncryptedInputFile> encrypted) {
     // Bulk decrypt is only applied to data files. Returning source input files for parquet.
-    return Iterables.transform(encrypted, this::decrypt);
+    if (nativeDataEncryption) {
+      return Iterables.transform(encrypted, this::getSourceFile);
+    } else {
+      return Iterables.transform(encrypted, this::decrypt);
+    }
+  }
+
+  private InputFile getSourceFile(EncryptedInputFile encryptedFile) {
```

Comment:
Style: Iceberg method names should not include `get`. Instead, use a more helpful verb like `create` or `find`, or simply leave it out.